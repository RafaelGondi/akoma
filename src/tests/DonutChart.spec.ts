import { describe, test, expect } from 'vitest';
import DonutChart from '../components/DonutChart.vue';
import { shallowMount } from '@vue/test-utils';

const mockedData =[
	{
		name: 'Ecocardiograma',
		datasets: [
			{
				label: 'Exames realizados',
				data: [120, 220, 180, 320, 150],
			},
		]
	},
];

const mockedColors = ['teal', 'violet', 'amber', 'blue', 'red'];

describe('DonutChart', () => {
	test('renders correctly', async () => {
		const wrapper = shallowMount(DonutChart, {
			props: {
				labels: ['Janeiro','Fevereiro','MarÃ§o', 'Abril', 'Maio'],
				variant: 'green',
				data: mockedData,
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	test('it accepts valid colors', () => {
		const validator = DonutChart.props.colors.validator;
		const invalidColors = [...mockedColors, 'silver'];
		expect(validator(mockedColors)).toBe(true);
		expect(validator(invalidColors)).toBe(false);
	});
});
