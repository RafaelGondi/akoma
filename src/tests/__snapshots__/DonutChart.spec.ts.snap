// Vitest Snapshot v1

exports[`DonutChart > renders correctly 1`] = `"<span data-v-2c35dc11=\\"\\"><div data-v-2c35dc11=\\"\\" class=\\"responsive-container\\"><doughnut-stub data-v-2c35dc11=\\"\\" data=\\"[object Object]\\" options=\\"[object Object]\\" plugins=\\"\\" datasetidkey=\\"label\\"></doughnut-stub></div></span>"`;
